### ISSUE TITLE
after 6.4.9-zen, tpm was broken

### ISSUE THREAD

**Issue by nerimoe** [zen-kernel/zen-kernel#323]
archlinux linux-zen 6.4.10
tpm2.0

![图片](https://github.com/zen-kernel/zen-kernel/assets/57009175/37b473a8-ff2b-4b6a-8a5a-976d8b51aa27)

and dmesg shows this message
`Aug 12 23:00:00 archlinux kernel: tpm_crb: probe of MSFT0101:00 failed with error 378`

**Comment by nerimoe** [comment]
with linux-lts
![图片](https://github.com/zen-kernel/zen-kernel/assets/57009175/9c6a4259-e5ee-4218-9cfc-20254fb1f07d)

**Comment by damentz** [comment]
You didn't provide information about your system, so I can only assume you're on AMD due to:

https://github.com/zen-kernel/zen-kernel/commit/4f58e962d7e1e2e43ed756b8cc3b745c672a8742
https://github.com/zen-kernel/zen-kernel/commit/78286766c890ebd64c2a52cff2ca41bd814277db

v6.4.11 is slated to remove fTPM support for hwrng, and also any current supported LTS kernels.

**Comment by damentz** [comment]
Closing, if you provide any useful info we can reopen if what I assumed is not the case.

**Comment by nerimoe** [comment]
intel 10710u, laptop

**Comment by nerimoe** [comment]
![图片](https://github.com/zen-kernel/zen-kernel/assets/57009175/563f55f0-1167-40f8-a1d3-c2605d30ae4c)
lenovo ideapad s540 with i7 10710u

**Comment by AzureZeng** [comment]
![image](https://github.com/zen-kernel/zen-kernel/assets/19504193/0c4118d9-e9c0-4ab9-8ea3-fd025f44f59f)
```$journalctl -k --grep=tpm
Aug 12 23:49:59 azw-arch-sys kernel: efi: ACPI=0x99c4d000 ACPI 2.0=0x99c4d014 TPMFinalLog=0x99b79000 ESRT=0x97de1018 SMBIOS=0x97de0000 SMBIOS 3.0=0x97dde000 MOKvar=0x97da3000 
Aug 12 23:49:59 azw-arch-sys kernel: ACPI: TPM2 0x0000000099C38000 000034 (v03 ACRSYS ACRPRDCT 00000002 1025 00040000)
Aug 12 23:49:59 azw-arch-sys kernel: ACPI: Reserving TPM2 table memory at [mem 0x99c38000-0x99c38033]
Aug 12 23:49:59 azw-arch-sys kernel: tpm_crb: probe of MSFT0101:00 failed with error 378
Aug 12 23:49:59 azw-arch-sys systemd[1]: systemd 254.1-1-arch running in system mode (+PAM +AUDIT -SELINUX -APPARMOR -IMA +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBFDISK +PCRE2 -PWQUALITY +P11KIT -QRENCODE +TPM2 +BZIP2 +LZ4 +XZ +ZLIB +ZSTD +BPF_FRAMEWORK +XKBCOMMON +UTMP -SYSVINIT default-hierarchy=unified)
Aug 12 23:49:59 azw-arch-sys systemd[1]: TPM2 PCR Machine ID Measurement was skipped because of an unmet condition check (ConditionPathExists=/sys/firmware/efi/efivars/StubPcrKernelImage-4a67b082-0a4c-41cf-b6c7-440b29bb8c4f).
```

```
# azurezeng @ azw-arch-sys in ~ [0:01:12] C:1
$ sudo dmesg | grep tpm_crb -A 10
[    0.398635] tpm_crb: probe of MSFT0101:00 failed with error 378
[    0.398652] AMD-Vi: AMD IOMMUv2 functionality not available on this system - This is not a bug.
[    0.398948] ahci 0000:00:17.0: version 3.0
[    0.409217] ahci 0000:00:17.0: AHCI 0001.0301 32 slots 1 ports 6 Gbps 0x1 impl SATA mode
[    0.409221] ahci 0000:00:17.0: flags: 64bit ncq sntf pm clo only pio slum part ems deso sadm sds apst 
[    0.409495] scsi host0: ahci
[    0.409553] ata1: SATA max UDMA/133 abar m2048@0xa1493000 port 0xa1493100 irq 126
[    0.409775] ACPI: bus type drm_connector registered
[    0.410337] usbcore: registered new interface driver usbserial_generic
[    0.410341] usbserial: USB Serial support registered for generic
[    0.410407] rtc_cmos 00:03: RTC can wake from S4
```

**Comment by damentz** [comment]
I see, seems to be unrelated to AMD, will reopen.

**Comment by damentz** [comment]
Per the comments in the merge and revert, the cherry picked fix may have been incomplete and depend on other code, so we'll just revert it and wait for the complete set of stable patches in v6.4.11.

This may not be the solution either, but it's the only way we deviated from upstream wrt TPM.

**Comment by Hazmi35** [comment]
Got the same problem in my Intel Laptop, my AMD Desktop doesn't have this issue. `linux` from Arch doesn't have the problem too, I also get the same `tpm_crb: probe of MSFT0101:00 failed with error 378` message

**Comment by nerimoe** [comment]
solved in 6.4.10-zen2
